### PR TITLE
Extending FAudioFilterParameters to include WetDryMix 

### PR DESCRIPTION
**Pull request by TylerGlaiel**
as discussed via twitter DM

Added FAudioFilterParametersEXT which includes WetDryMix as a field and EXT variations of the appropriate get and set functions.

Internally, all instances of FAudioFilterParameters were swapped to FAudioFilterParametersEXT, the non-EXT versions of the get and set functions convert FAudioFilterParameters to the EXT version (with the default WetDryMix value of 1) then redirect to the EXT versions of the functions.

### CODE REVIEW

**Comment by TylerGlaiel** [comment]
Possible concerns:

FACT_internal also uses FAudioFilterParamers in there, I did not change anything in that file as it didn't seem to touch anything in FAudio_internal

File: include/FAudio.h [hunk 1]
```diff
@@ -375,6 +383,7 @@ typedef struct FAudioXMA2WaveFormatEx
 #define FAUDIO_DEFAULT_FILTER_TYPE	FAudioLowPassFilter
 #define FAUDIO_DEFAULT_FILTER_FREQUENCY	FAUDIO_MAX_FILTER_FREQUENCY
 #define FAUDIO_DEFAULT_FILTER_ONEOVERQ	1.0f
+#define FAUDIO_DEFAULT_FILTER_WETDRYMIX	1.0f
```
Comment (flibitijibibo) [inline include/FAudio.h:386]:
```suggestion
#define FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT	1.0f
```

File: src/FAudio.c [hunk 2]
```diff
@@ -278,6 +278,7 @@ uint32_t FAudio_CreateSourceVoice(
 	(*ppSourceVoice)->filter.Type = FAUDIO_DEFAULT_FILTER_TYPE;
 	(*ppSourceVoice)->filter.Frequency = FAUDIO_DEFAULT_FILTER_FREQUENCY;
 	(*ppSourceVoice)->filter.OneOverQ = FAUDIO_DEFAULT_FILTER_ONEOVERQ;
+	(*ppSourceVoice)->filter.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX;
```
Comment (flibitijibibo) [inline src/FAudio.c:281]:
```suggestion
	(*ppSourceVoice)->filter.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT;
```

File: src/FAudio.c [hunk 3]
```diff
@@ -586,6 +587,7 @@ uint32_t FAudio_CreateSubmixVoice(
 	(*ppSubmixVoice)->filter.Type = FAUDIO_DEFAULT_FILTER_TYPE;
 	(*ppSubmixVoice)->filter.Frequency = FAUDIO_DEFAULT_FILTER_FREQUENCY;
 	(*ppSubmixVoice)->filter.OneOverQ = FAUDIO_DEFAULT_FILTER_ONEOVERQ;
+	(*ppSubmixVoice)->filter.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX;
```
Comment (flibitijibibo) [inline src/FAudio.c:590]:
```suggestion
	(*ppSubmixVoice)->filter.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT;
```

File: src/FAudio.c [hunk 4]
```diff
@@ -1315,6 +1317,7 @@ uint32_t FAudioVoice_SetOutputVoices(
 			voice->sendFilter[i].Type = FAUDIO_DEFAULT_FILTER_TYPE;
 			voice->sendFilter[i].Frequency = FAUDIO_DEFAULT_FILTER_FREQUENCY;
 			voice->sendFilter[i].OneOverQ = FAUDIO_DEFAULT_FILTER_ONEOVERQ;
+			voice->sendFilter[i].WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX;
```
Comment (flibitijibibo) [inline src/FAudio.c:1320]:
```suggestion
			voice->sendFilter[i].WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT;
```

File: src/FAudio.c [hunk 5]
```diff
@@ -1687,9 +1690,23 @@ uint32_t FAudioVoice_SetFilterParameters(
...
+uint32_t FAudioVoice_SetFilterParameters(
+	FAudioVoice* voice,
+	const FAudioFilterParameters* pParameters,
+	uint32_t OperationSet
+) {
+	FAudioFilterParametersEXT ext_parameters;
+	ext_parameters.Type = pParameters->Type;
+	ext_parameters.OneOverQ = pParameters->OneOverQ;
+	ext_parameters.Frequency = pParameters->Frequency;
+	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX;
```
Comment (flibitijibibo) [inline src/FAudio.c:1702]:
```suggestion
	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT;
```

File: src/FAudio.c [hunk 6]
```diff
@@ -1713,17 +1730,34 @@ void FAudioVoice_GetFilterParameters(
...
-uint32_t FAudioVoice_SetOutputFilterParameters(
+void FAudioVoice_GetFilterParameters(
+	FAudioVoice* voice,
+	FAudioFilterParameters* pParameters
+) {
+	FAudioFilterParametersEXT ext_parameters;
+	ext_parameters.Type = pParameters->Type;
+	ext_parameters.OneOverQ = pParameters->OneOverQ;
+	ext_parameters.Frequency = pParameters->Frequency;
+	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX;
```
Comment (flibitijibibo) [inline src/FAudio.c:1748]:
```suggestion
	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT;
```

File: src/FAudio.c [hunk 7]
```diff
@@ -1800,10 +1834,25 @@ uint32_t FAudioVoice_SetOutputFilterParameters(
...
+	FAudioVoice* voice,
+	FAudioVoice* pDestinationVoice,
+	const FAudioFilterParameters* pParameters,
+	uint32_t OperationSet
+) {
+	FAudioFilterParametersEXT ext_parameters;
+	ext_parameters.Type = pParameters->Type;
+	ext_parameters.OneOverQ = pParameters->OneOverQ;
+	ext_parameters.Frequency = pParameters->Frequency;
+	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX;
```
Comment (flibitijibibo) [inline src/FAudio.c:1847]:
```suggestion
	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT;
```

File: src/FAudio.c [hunk 8]
```diff
@@ -1914,6 +1963,24 @@ uint32_t FAudioVoice_SetVolume(
...
+void FAudioVoice_GetOutputFilterParameters(
+	FAudioVoice* voice,
+	FAudioVoice* pDestinationVoice,
+	FAudioFilterParameters* pParameters
+) {
+	FAudioFilterParametersEXT ext_parameters;
+	ext_parameters.Type = pParameters->Type;
+	ext_parameters.OneOverQ = pParameters->OneOverQ;
+	ext_parameters.Frequency = pParameters->Frequency;
+	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX;
```
Comment (flibitijibibo) [inline src/FAudio.c:1975]:
```suggestion
	ext_parameters.WetDryMix = FAUDIO_DEFAULT_FILTER_WETDRYMIX_EXT;
```

**Review by flibitijibibo — COMMENTED** [review]
Just needed to add `_EXT` to the define, otherwise lgtm

**Comment by flibitijibibo** [comment]
> FACT_internal also uses FAudioFilterParamers in there, I did not change anything in that file as it didn't seem to touch anything in FAudio_internal

Nah, FACT's supposed to use all-public features without FAudio_internal stuff (aside from stdlib maybe)

> Not doing any bounds checking or clamping for WetDryMix internally (do we need to?)

Actually no; XAudio2 notoriously doesn't do bounds checking for most volume/mix-related values, you can even do negative numbers for things like volume and it's completely intentional!

> double check all the places where it's getting memcpy'd

Looks like they were all found, I usually do explicit sizeof() so if we missed it somewhere that's on me...

**Comment by flibitijibibo** [comment]
Will write up the extension spec on Monday, thanks!

**Pull request merged**